Auth0/SafariAuthenticationSession.swift

```diff
@@ -22,31 +22,39 @@
 
 import UIKit
 import SafariServices
+import AuthenticationServices
 
 #if swift(>=3.2)
 @available(iOS 11.0, *)
 class SafariAuthenticationSession: AuthSession {
 
-    typealias FinishSession = (Result<Credentials>) -> Void
-
-    var authSession: SFAuthenticationSession?
-    let authorizeURL: URL
-
     init(authorizeURL: URL, redirectURL: URL, state: String? = nil, handler: OAuth2Grant, finish: @escaping FinishSession, logger: Logger?) {
-        self.authorizeURL = authorizeURL
         super.init(redirectURL: redirectURL, state: state, handler: handler, finish: finish, logger: logger)
-        self.authSession = SFAuthenticationSession(url: self.authorizeURL, callbackURLScheme: self.redirectURL.absoluteString) { [unowned self] in
```

Comment:
Retention doesn't appear to be an issue here, although I'm sure it was hence the use of the `authSession` property.
It's difficult to test these behaviour in unit tests.

Reply:
Looks like SFAuthenticationSession requires a strong reference, as per the apple docs

> Note
> 
> Ensure that there is a strong reference to the SFAuthenticationSession instance when the session is in progress.

I don't see a similar note for ASWebAuthenticationSession

Reply:
If ASWebAuthenticationSession is deallocated, the permissions dialog is dismissed